lib/validator/validator.go

```diff
@@ -36,22 +36,28 @@ import (
 var validate *validator.Validate
 
 var (
-	labelRegex          = regexp.MustCompile(`^` + tokenizer.LabelKeyMatcher + `$`)
-	labelValueRegex     = regexp.MustCompile("^[a-zA-Z0-9]?([a-zA-Z0-9_.-]{0,61}[a-zA-Z0-9])?$")
-	nameRegex           = regexp.MustCompile("^[a-zA-Z0-9_.-]{1,128}$")
-	namespacedNameRegex = regexp.MustCompile(`^[a-zA-Z0-9_./-]{1,128}$`)
-	interfaceRegex      = regexp.MustCompile("^[a-zA-Z0-9_-]{1,15}$")
-	actionRegex         = regexp.MustCompile("^(allow|deny|log|pass)$")
-	backendActionRegex  = regexp.MustCompile("^(allow|deny|log|next-tier|)$")
-	protocolRegex       = regexp.MustCompile("^(tcp|udp|icmp|icmpv6|sctp|udplite)$")
-	ipipModeRegex       = regexp.MustCompile("^(always|cross-subnet|)$")
-	reasonString        = "Reason: "
-	poolSmallIPv4       = "IP pool size is too small (min /26) for use with Calico IPAM"
-	poolSmallIPv6       = "IP pool size is too small (min /122) for use with Calico IPAM"
-	poolUnstictCIDR     = "IP pool CIDR is not strictly masked"
-	overlapsV4LinkLocal = "IP pool range overlaps with IPv4 Link Local range 169.254.0.0/16"
-	overlapsV6LinkLocal = "IP pool range overlaps with IPv6 Link Local range fe80::/10"
-	protocolPortsMsg    = "rules that specify ports must set protocol to TCP or UDP"
+	labelRegex            = regexp.MustCompile(`^` + tokenizer.LabelKeyMatcher + `$`)
+	labelValueRegex       = regexp.MustCompile("^[a-zA-Z0-9]?([a-zA-Z0-9_.-]{0,61}[a-zA-Z0-9])?$")
+	nameRegex             = regexp.MustCompile("^[a-zA-Z0-9_.-]{1,128}$")
```

Comment:
name regex should match the k8s one.

and in addition, Calico Network policies names cannot have a "." in them unless they start with "knp.default."

Reply:
do you mean the `metadata` name field? we're not currently validating metadata.

Reply:
we should be validating the metadata.  I think there is a subtask for that no?  Or did that go to me?  I forget now.

Reply:
The only things we *have* to validate in the Metadata are the name, labels and annotations.

All of the other stuff is only configurable through the  AAPIS, so we don't really need to validate it here.

Reply:
we discussed addressing metadata in a subsequent PR since it's not clear how to use our current approach with the k8s `Metadata` types. The better longer term approach is to adopt `apimachinery` validation where appropriate.